Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,72 @@ namespace TestImpact
+            AZStd::map<AZStd::string, AZStd::map<AZStd::string, AZStd::set<AZStd::string>>> testMapping =
+                AZStd::map<AZStd::string, AZStd::map<AZStd::string, AZStd::set<AZStd::string>>>();
+
+            // Steps through our string line by line. Assumes lines are split by newline characters.
+            while ((startLine = testEnumerationData.find_first_not_of(newLineDelim, endLine)) != AZStd::string::npos)
+            {
+                endLine = testEnumerationData.find(newLineDelim, startLine);
+                AZStd::string curLine = testEnumerationData.substr(startLine, endLine - startLine);
+
+                curLine = curLine.substr(0, curLine.find_first_of('['));
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                AZStd::basic_regex testNamePattern = AZStd::basic_regex("([a-zA-Z_/\\-0-9.]*\\.py)::([A-Za-z_/\\-0-9]*)::(.*)");
+                AZStd::smatch matchResults;
+                if (AZStd::regex_search(curLine, matchResults, testNamePattern))
+                {
+                    AZStd::string absoluteScriptPath = matchResults[1];
+                    AZStd::string testFixture = matchResults[2];
+                    AZStd::string testName = matchResults[3];
+
+                    // Fetch or create the map for this script
+                    AZStd::map<AZStd::string, AZStd::set<AZStd::string>>& moduleMap = testMapping[absoluteScriptPath];
+
+                    // Fetch or create the set for this test fixture, then add our test name to the set.
+                    AZStd::set<AZStd::string>& testNameSet = moduleMap[testFixture];
+                    testNameSet.insert(testName);
+                }
+            }
+
+            // Create a list of test suites for each script
+            // Lookup the map for that script path, and for each test class in that map, create a test suite object and add all the tests to
+            // that test suite. Create a pair consisting of the script path and the test suites attached to that script, and add to our
+            // vector.
+            for (const auto& [scriptPath, moduleMap] : testMapping)
+            {
+                AZStd::vector<TestEnumerationSuite> testSuites = AZStd::vector<TestEnumerationSuite>();
+                for (const auto& [moduleName, testSet] : moduleMap)
+                {
+                    TestEnumerationSuite testSuite;
+                    testSuite.m_name = moduleName;
+                    testSuite.m_enabled = true;
+                    for (const AZStd::string testName : testSet)
+                    {
+                        TestEnumerationCase testCase;
+                        testCase.m_name = testName;
+                        testCase.m_enabled = true;
+                        testSuite.m_tests.emplace_back(AZStd::move(testCase));
```

Comment:
This should be `testSuite.m_tests.emplace_back(testName, true);`

Reply:
Have had to use TestEnumerationCase { testName, true} and TestEnumerationSuite{testFixture, true, newVector} due to compile issues, but has been simplified a bit.